### PR TITLE
Document behavior of LIMIT on side effects

### PR DESCRIPTION
**Pull request by loveleif**

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala [hunk 1]
```diff
@@ -76,6 +78,36 @@ class LimitTest extends DocumentingTest {
         resultTable()
       }
     }
+    section("`LIMIT` will not stop side effects", "limit-will-not-stop-side-effects") {
+      p(
+        """The use of LIMIT in a query will not stop side effects, like `CREATE`, `DELETE` or `SET`, from happening. This behaviour was undefined in versions before 4.3.""".stripMargin
```
Comment (pontusmelke) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala]: Maybe add a sentence and example saying that if you want to control updates you need to do something like 

```
MATCH (n {name: 'A'})
WITH n LIMIT 0
SET n.age = 60
RETURN n
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala [hunk 2]
```diff
@@ -76,6 +78,54 @@ class LimitTest extends DocumentingTest {
...
+      query(
+        """CREATE (n)
+          |RETURN n
+          |LIMIT 0""".stripMargin,
+        ResultAssertions((r) => {
+          r.size shouldBe 0
+          assertStats(r, nodesCreated = 1)
+        })
+      ) {
+        p("Returns nothing but creates one node.")
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala]:
```suggestion
        p("This query returns nothing, but creates one node:")
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala [hunk 3]
```diff
@@ -76,6 +78,54 @@ class LimitTest extends DocumentingTest {
...
+        """MATCH (n {name: 'A'})
+          |SET n.age = 60
+          |RETURN n
+          |LIMIT 0""".stripMargin,
+        ResultAssertions((r) => {
+          r.size shouldBe 0
+          assertStats(r, propertiesWritten = 1)
+        })
+      ) {
+        p("Returns nothing but writes one property.")
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala]:
```suggestion
        p("This query returns nothing, but writes one property:")
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala [hunk 4]
```diff
@@ -76,6 +78,54 @@ class LimitTest extends DocumentingTest {
...
+        resultTable()
+      }
+      p(
+        "If we want to limit the number of updates we can split the query using the WITH clause."
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala]:
```suggestion
        "If we want to limit the number of updates we can split the query using the `WITH` clause:"
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala [hunk 5]
```diff
@@ -76,6 +78,54 @@ class LimitTest extends DocumentingTest {
         resultTable()
       }
     }
+    section("`LIMIT` will not stop side effects", "limit-will-not-stop-side-effects") {
+      p(
+        """The use of LIMIT in a query will not stop side effects, like `CREATE`, `DELETE` or `SET`, from happening if
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala]:
```suggestion
        """The use of `LIMIT` in a query will not stop side effects, like `CREATE`, `DELETE` or `SET`, from happening if
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala [hunk 6]
```diff
@@ -76,6 +78,54 @@ class LimitTest extends DocumentingTest {
...
+        """MATCH (n)
+          |WITH n LIMIT 1
+          |SET n.locked = true
+          |RETURN n""".stripMargin,
+        ResultAssertions((r) => {
+          r.size shouldBe 1
+          assertStats(r, propertiesWritten = 1)
+        })
+      ) {
+        p("Writes `locked` property on one node and return that node.")
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/LimitTest.scala]:
```suggestion
        p("Writes `locked` property on one node and return that node:")
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @loveleif 👍 

I've pushed some minor editorial updates

**Comment by davidoliverSP2** [comment]
Hi @loveleif - there appears to be some test problems related to this PR. Could you take a look please?

**Comment by loveleif** [comment]
> there appears to be some test problems related to this PR. Could you take a look please?

@davidoliverSP2 some unrelated tests were failing, rebasing the branch made that go away.

**Pull request merged**